### PR TITLE
Update ground detect

### PR DESCRIPTION
**Pull request by dewagter**

### CODE REVIEW

File: conf/modules/ground_detect.xml [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
   </doc>
   <header>
-    <file name="ground_detect_sensor.h"/>
+    <file name="ground_detect.h"/>
   </header>
-  <init fun="ground_detect_sensor_init()"/>
-  <periodic fun="ground_detect_sensor_periodic()" freq="50"/>
+  <init fun="ground_detect_init()"/>
+  <periodic fun="ground_detect_periodic()" freq="50"/>
+  <periodic fun="ground_detect_filter_accel()" freq="500"/>
```
[the remark this answers is not in the document]
Comment (dewagter) [reply conf/modules/ground_detect.xml:18]:
```suggestion
  <periodic fun="ground_detect_filter_accel()"/>
```
Agreed!

File: conf/airframes/tudelft/rotwing5.xml [hunk 2]
```diff
@@ -255,7 +255,7 @@
         <define name="CHIRP_RADIO_CHANNEL" value="9"/>
     </section>
 
-    <section name="CTRL_EFF_SHED" prefix="ROTWING_EFF_SCHED_">
+    <section name="CTRL_EFF_SHED" prefix="ROT_WING_EFF_SCHED_">
```
Comment (gautierhattenberger) [inline conf/airframes/tudelft/rotwing5.xml:258]: Are you sure about the name ? I think you changed them all to `ROTWING_`

File: conf/airframes/tudelft/rotwing_v3b.xml [hunk 3]
```diff
@@ -296,7 +296,7 @@
         <define name="CHIRP_RADIO_CHANNEL" value="9"/>
     </section>
 
-    <section name="CTRL_EFF_SHED" prefix="ROTWING_EFF_SCHED_">
+    <section name="CTRL_EFF_SHED" prefix="ROT_WING_EFF_SCHED_">
```
Comment (gautierhattenberger) [inline conf/airframes/tudelft/rotwing_v3b.xml:299]: same here

**Review by gautierhattenberger — DISMISSED** [review]

File: conf/modules/ground_detect.xml [hunk 4]
```diff
@@ -1,23 +1,26 @@
...
+  <init fun="ground_detect_init()"/>
+  <periodic fun="ground_detect_periodic()" freq="50"/>
+  <periodic fun="ground_detect_filter_accel()"/>
   <makefile>
-    <file name="ground_detect_sensor.c"/>
-    <test>
+    <file name="ground_detect.c"/>
+    <test firmware="rotorcraft">
+      <define name="PERIODIC_FREQUENCY" value="512"/>
+      <define name="PERIODIC_TELEMETRY" value="0"/>
```
Comment (gautierhattenberger) [inline conf/modules/ground_detect.xml:23]: `PERIODIC_TELEMETRY` is enabled for tests, so adding it here makes a `redefined` warning and doesn't disable it. Instead add something like:
```
      <define name="DOWNLINK_TRANSPORT" value="pprz_tp"/>
      <define name="DOWNLINK_DEVICE" value="uart0"/>
      <define name="USE_UART0"/>
```

File: sw/airborne/modules/nav/ground_detect.c [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  }
+
+#ifdef DEBUG_GROUND_DETECT
+  float payload[7];
+  payload[0] = vspeed_ned;
+  payload[1] = spec_thrust_down;
+  payload[2] = accel_filter.o[0];
+  payload[3] = stateGetAccelNed_f()->z;
+  payload[4] = agl_dist_valid;
+  payload[5] = agl_dist_value_filtered;
```
Comment (gautierhattenberger) [inline sw/airborne/modules/nav/ground_detect.c:133]: These lines should be protected with `#if USE_GROUND_DETECT_AGL_DIST`

**Review by gautierhattenberger — CHANGES_REQUESTED** [review]

**Review by gautierhattenberger — APPROVED** [review]

**Pull request merged**